### PR TITLE
fix: Event Grid System Topic Event Subscription w/ Managed Identity #705

### LINKED ISSUE #705: [AVM Module Issue]: Event Grid System Topic Event Subscription w/ Managed Identity
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/event-grid/system-topic

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

_No response_

### Description

https://github.com/Azure/bicep/blob/4fff3423026fa5284e1effc51afbba1a0ecd6ec7/src/Bicep.Core.IntegrationTests/ScenarioTests.cs#L2298-L2299

When you use managed identity for the destination you never set destination. Bicep requires this field to be passed.

```json
{
  "code": "InvalidRequest",
  "message": "Destination and DeliveryWithResourceIdentity cannot be set at the same time."
}
```

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by fabmas**
## Description

fixed the Issue 
Closes #705 

[avm.res.event-grid.system-topic](https://github.com/fabmas/bicep-registry-modules/actions/workflows/avm.res.event-grid.system-topic.yml)

## Updating an existing module

- [ x ] This is a bug fix:
  - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.
- [x] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [x] I have updated the examples in README with the latest module version number.

### CODE REVIEW

File: avm/res/event-grid/system-topic/event-subscription/main.bicep [hunk 1]
```diff
@@ -72,7 +72,7 @@ resource eventSubscription 'Microsoft.EventGrid/systemTopics/eventSubscriptions@
     deadLetterDestination: !empty(deadLetterDestination) ? deadLetterDestination : null
     deadLetterWithResourceIdentity: !empty(deadLetterWithResourceIdentity) ? deadLetterWithResourceIdentity : null
     deliveryWithResourceIdentity: !empty(deliveryWithResourceIdentity) ? deliveryWithResourceIdentity : null
-    destination: destination
+    destination: !empty(deliveryWithResourceIdentity) ? null: destination
```
Comment (AlexanderSehr) [inline avm/res/event-grid/system-topic/event-subscription/main.bicep:75]:
```suggestion
    destination: empty(deliveryWithResourceIdentity) ? destination : null
```
Always a fan of having the `null` as an else as it's easier to read. Note - requires a regeneration of the `main.json`

File: avm/res/event-grid/topic/event-subscription/main.bicep [hunk 2]
```diff
@@ -54,7 +54,7 @@ resource eventSubscription 'Microsoft.EventGrid/topics/eventSubscriptions@2022-0
     deadLetterDestination: !empty(deadLetterDestination) ? deadLetterDestination : null
     deadLetterWithResourceIdentity: !empty(deadLetterWithResourceIdentity) ? deadLetterWithResourceIdentity : null
     deliveryWithResourceIdentity: !empty(deliveryWithResourceIdentity) ? deliveryWithResourceIdentity : null
-    destination: destination
+    destination: !empty(deliveryWithResourceIdentity) ? null: destination
```
Comment (AlexanderSehr) [inline avm/res/event-grid/topic/event-subscription/main.bicep:57]:
```suggestion
    destination: empty(deliveryWithResourceIdentity) ? destination : null
```
Same as the other comment :)

Comment (fabmas) [reply avm/res/event-grid/topic/event-subscription/main.bicep]: OK !

File: avm/res/event-grid/system-topic/event-subscription/main.bicep (hunk 1 above)
Comment (fabmas) [reply avm/res/event-grid/system-topic/event-subscription/main.bicep]: OK !

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**